src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -19,21 +18,27 @@ public class CommandResult {
     /** The application should exit. */
     private final boolean exit;
 
+    private final boolean showPerson;
+    private final Person viewedPerson;
+
     /**
      * Constructs a {@code CommandResult} with the specified fields.
      */
-    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
-        this.feedbackToUser = requireNonNull(feedbackToUser);
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit,
+                         boolean showPerson, Person viewedPerson) {
+        this.feedbackToUser = feedbackToUser;
         this.showHelp = showHelp;
         this.exit = exit;
+        this.showPerson = showPerson;
+        this.viewedPerson = viewedPerson;
     }
 
     /**
      * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
      * and other fields set to their default value.
      */
     public CommandResult(String feedbackToUser) {
-        this(feedbackToUser, false, false);
+        this(feedbackToUser, false, false, false, null);
```

Comment:
Instead of using null as a place holder value, perhaps we could consider using a static constant EMPTY_PERSON to align with OOP principles?

Reply:
Adding on, you can place this `EMPTY_PERSON` or `DEFAULT_PERSON` value in `CommandCommons` class.